--- libosip2-5.2.1/configure.ac.orig	2021-06-12 17:50:24.019202166 +0200
+++ libosip2-5.2.1/configure.ac	2021-06-12 17:55:50.656148184 +0200
@@ -187,11 +187,11 @@
 AM_CONDITIONAL(BUILD_MAXSIZE, test x$enable_minisize = xno)
 
 dnl Checks for libraries. (those one are needed for sun)
-AC_CHECK_LIB(posix4,sem_open,[FSM_LIB="$FSM_LIB -lposix4 -mt"])
-AC_CHECK_LIB(nsl,nis_add,[FSM_LIB="$FSM_LIB -lnsl"])
-AC_CHECK_LIB(socket,sendto,[FSM_LIB="$FSM_LIB -lsocket"])
+AC_CHECK_FUNC(sem_open, , [AC_CHECK_LIB(posix4,sem_open,[FSM_LIB="$FSM_LIB -lposix4 -mt"])])
+AC_CHECK_FUNC(nis_add, , [AC_CHECK_LIB(nsl,nis_add,[FSM_LIB="$FSM_LIB -lnsl"])])
+AC_CHECK_FUNC(sendto, , [AC_CHECK_LIB(socket,sendto,[FSM_LIB="$FSM_LIB -lsocket"])])
 AC_CHECK_LIB(c, lrand48, AC_DEFINE([HAVE_LRAND48], [], [Defined if you have lrand48]))
-AC_CHECK_LIB(rt,clock_gettime,[FSM_LIB="$FSM_LIB -lrt"])
+AC_CHECK_FUNC(clock_gettime, , [AC_CHECK_LIB(rt,clock_gettime,[FSM_LIB="$FSM_LIB -lrt"])])
 
 AC_CHECK_FUNCS([localtime])
 
